go/vt/vttablet/grpctmclient/client.go

```diff
@@ -630,6 +684,9 @@ func (client *Client) FullStatus(ctx context.Context, tablet *topodatapb.Tablet)
 
 	response, err := c.FullStatus(ctx, &tabletmanagerdatapb.FullStatusRequest{})
 	if err != nil {
+		if invalidator != nil {
+			invalidator()
+		}
 		return nil, err
```

Comment:
nit: this will not be nil

Reply:
It could be `nil`, if this condition fails:
```go
if poolDialer, ok := client.dialer.(poolDialer); ok
```

Now, I'm not sure how it could possibly fail, but as golang goes, I created `var invalidator invalidatorFunc` and it could be `nil`.

Reply:
If you like, I can pre-assign `var invalidator` to an empty function? That way it will never be `nil`.

Reply:
This kinda returns to https://github.com/vitessio/vitess/pull/14979#discussion_r1456927968